### PR TITLE
Remove ruby execution engine

### PR DESCRIPTION
**Pull request by kaisecheng**
Fixed: #11236

This PR removes the Ruby Execution Engine
The Java Execution Engine has been the default engine since Logstash 7.0, and is capable of using plugins written in either Ruby or Java; removing the Ruby Execution Engine will not affect our ability to run existing pipelines, but will greatly reduce the surface area of all features and fixes within the engine(s). We will no longer need to target two engines simultaneously.

Ruby execution engine uses Treetop to parse Logstash config files and convert them to executable code which is being referenced by Java pipeline, so Treetop stays in the project.

- [x] remove Logstash::Pipeline
- [x] rewrite test cases which depend on ruby pipeline
- [x] remove logstash/event, Logstash::SignalEvent
- [x] remove Engine.Ruby on java side
- [x] remove config pipeline.java_execution

This PR will split into three smaller to feature branch for review
#12496 remove Logstash::Pipeline, Engine.Ruby, rewrite test cases
#12505 remove Logstash::SignalEvent
#12506 remove config `pipeline.java_execution`

A follow up will be removing logstash/pipeline in `logstash-devutils`
- [x] clean up devutils https://github.com/elastic/logstash-devutils/pull/89
- [x] check tier 1 plugins reference
- [x] clean up input-redis https://github.com/logstash-plugins/logstash-input-redis/pull/84
- [x] check tier 2 plugins reference

### CODE REVIEW

**Review by andsel — APPROVED** [review]
I think the commits should be squashed, LGTM

File: logstash-core/lib/logstash/pipeline.rb [hunk 1]
```diff
@@ -15,681 +15,24 @@
...
-require "logstash/filters/base"
-require "logstash/inputs/base"
-require "logstash/outputs/base"
-require "logstash/instrument/collector"
-require "logstash/filter_delegator"
-require "logstash/compiler"
+# for backward compatibility
+# logstash-devutils-1.3.6 logstash_helpers has dependency on this class
+module LogStash
+  class Pipeline
```
Comment (kares) [inline logstash-core/lib/logstash/pipeline.rb:21]: NOTE: any plugin doing `require 'logstash/pipeline'` (e.g. in specs) will be broken by this.
Wonder if it was considered to have this file around as:
```ruby
require 'logstash/java_pipeline'
module LogStash
  Pipeline = JavaPipeline
end
```

Comment (kaisecheng) [reply logstash-core/lib/logstash/pipeline.rb:21]: This is a nice suggestion. I was thinking to remove everything completely including the plugins.

File: docs/static/running-logstash-command-line.asciidoc [hunk 2]
```diff
@@ -101,10 +101,6 @@ With this command, Logstash concatenates three config files, `/tmp/one`, `/tmp/t
   If you wish to use both defaults, please use the empty string for the `-e` flag.
   The default is nil.
 
-*`--java-execution`*::
```
Comment (yaauie) [inline docs/static/running-logstash-command-line.asciidoc:104]: Latergram:

I think we can do to make this transition easier for our users, especially for those who opt for explicit configuration or who control their configuration with environment variables.

 - `7.x`: the `false` value for `--java-execution` or `java.execution` is deprecated. We emit a deprecation log that REE is deprecated and will be removed in "a future major release of Logstash". To solve this deprecation warning, the user can either remove the flag or set its value to `true`.
 - `8.x`: the `--java-execution` or `java.execution` option _itself_ is deprecated, and only accepts the `true` value. When this is specified, we emit a deprecation log about the option being removed in "a future major release of Logstash"

The primary goal that this path accomplishes is that explicit configuration from Logstash `7.$last` that contains no deprecations works without modification in `8.0.0`. It allows our users to be explicit in their configuration of `7.x` and does not require an intermediate step where they are required to remove settings from their config that are meaningful in Logstash 7 prior to upgrading to Logstash 8.

The other transition option is to fast-track the removal of the setting, but I think that may be less clear to the user.

 - `7.x`: the `--java-execution` or `java.execution` option _itself_ is deprecated.
   - When this option is specified as `true`, we emit a deprecation log indicating that it is not needed and the option will be removed in "a future major release of Logstash".
   - When the option is specified as `false`, we emit a different deprecation log indicating that both REE and the option will be removed in "a future major release of Logstash".

Note that the fast-track will make environment variable control of the config file not possible.

**Pull request merged**